controller/controller.go

```diff
@@ -65,6 +67,14 @@ func DefaultConfig() Config {
 // Controller defines the interface a controller needs to implement to provide
 // operations for groups of unit files against a fleet cluster.
 type Controller interface {
+	// GroupNeedsUpdate checks if the given group should be updated or not. To
+	// make a decision the unit content of each unit of each slice is compared
+	// using its unit hash. As soon as one unit hash differs, or a unit cannot be
+	// found, Inago assumes the whole group slice to be "dirty" and returns true
+	// having the group slices removed from the given req that are up to date,
+	// otherwise false, leaving the req as it is.
+	GroupNeedsUpdate(req Request) (Request, bool, error)
```

Comment:
1) The comment states this.

> and returns true having the group slices removed from the given req that are up to date

The slices that are up to date will be removed. The slices that need to be updated stay in there. 

> Can't we just return the dirty slices?

Sure. That makes more sense. 

2) You are asking over and over again why this and that should be in the public interface. We already talked about this. Lets have a talk again so we don't need to talk about it in every pull request.